File: library/src/test/java/io/github/anderscheow/validator/rules/regex/DigitsRuleTest.java
package io.github.anderscheow.validator.rules.regex;

import android.support.annotation.StringRes;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

public class DigitsRuleTest {

    private DigitsRule digitsRule;

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
        digitsRule = null;
    }

    @Test(expected = NullPointerException.class)
    public void validate_EmptySample_ThrowNullPointerException() throws Exception {
        digitsRule = new DigitsRule();

        digitsRule.validate(null);
    }

    @Test
    public void validate_SampleWithDigitsOnly_ReturnTrue() throws Exception {
        digitsRule = new DigitsRule();

        String sample = "123";

        assertTrue(digitsRule.validate(sample));
    }

    @Test
    public void validate_SampleWithAlphanumeric_ReturnFalse() throws Exception {
        digitsRule = new DigitsRule();

        String sample = "abc123";

        assertFalse(digitsRule.validate(sample));
    }

    @Test
    public void validate_SampleWithAlphabetOnly_ReturnFalse() throws Exception {
        digitsRule = new DigitsRule();

        String sample = "abc";

        assertFalse(digitsRule.validate(sample));
    }

    @Test
    public void validate_SampleWithDigitsAndSymbols_ReturnFalse() throws Exception {
        digitsRule = new DigitsRule();

        String sample = "100,000,000";

        assertFalse(digitsRule.validate(sample));
    }

    @Test
    public void errorMessage_DefaultErrorMessage() throws Exception {
        String errorMessage = "Value does not match digits regex";

        digitsRule = new DigitsRule();

        assertEquals(errorMessage, digitsRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorMessage() throws Exception {
        String errorMessage = "This is a custom error message";

        digitsRule = new DigitsRule(errorMessage);

        assertEquals(errorMessage, digitsRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorRes() throws Exception {
        @StringRes int errorRes = 0;

        digitsRule = new DigitsRule(errorRes);

        assertEquals(errorRes, digitsRule.errorRes());
    }
}

File: library/src/test/java/io/github/anderscheow/validator/rules/regex/DigitsRuleTest.kt
package io.github.anderscheow.validator.rules.regex

import android.support.annotation.StringRes
import org.junit.After
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class DigitsRuleTest {

    private lateinit var digitsRule: DigitsRule

    @Before
    @Throws(Exception::class)
    fun setUp() {
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @Test(expected = NullPointerException::class)
    @Throws(Exception::class)
    fun validate_EmptySample_ThrowNullPointerException() {
        digitsRule = DigitsRule()

        digitsRule.validate(null)
    }

    @Test
    @Throws(Exception::class)
    fun validate_SampleWithDigitsOnly_ReturnTrue() {
        digitsRule = DigitsRule()

        val sample = "123"

        assertTrue(digitsRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_SampleWithAlphanumeric_ReturnFalse() {
        digitsRule = DigitsRule()

        val sample = "abc123"

        assertFalse(digitsRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_SampleWithAlphabetOnly_ReturnFalse() {
        digitsRule = DigitsRule()

        val sample = "abc"

        assertFalse(digitsRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_SampleWithDigitsAndSymbols_ReturnFalse() {
        digitsRule = DigitsRule()

        val sample = "100,000,000"

        assertFalse(digitsRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_DefaultErrorMessage() {
        val errorMessage = "Value does not match digits regex"

        digitsRule = DigitsRule()

        assertEquals(errorMessage, digitsRule.getErrorMessage())
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorMessage() {
        val errorMessage = "This is a custom error message"

        digitsRule = DigitsRule(errorMessage)

        assertEquals(errorMessage, digitsRule.getErrorMessage())
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorRes() {
        @StringRes val errorRes = 0

        digitsRule = DigitsRule(errorRes)

        assertEquals(errorRes, digitsRule.getErrorRes())
    }
}